File: docs/examples/rs/examples/buckets_remove.rs
use reduct_rs::{ErrorCode, ReductClient, ReductError};
use tokio;

#[tokio::main]
async fn main() -> Result<(), ReductError> {
    // Create a new client with the API URL and API token
    let client = ReductClient::builder()
        .url("http://127.0.0.1:8383")
        .api_token("my-token")
        .build();

    // Remove the bucket "bucket-to-remove"
    let bucket = client.get_bucket("bucket-to-remove").await?;
    bucket.remove().await?;

    // Check that the bucket no longer exists or is in the process of being removed
    let status_code = client.get_bucket("bucket-to-remove").await.err().unwrap().status;
    assert_eq!(
        status_code,
        ErrorCode::NotFound | ErrorCode::Conflict
    );
    Ok(())
}

File: versioned_docs/version-1.20.x/examples/go/src/buckets_remove.go
package main

import (
	"context"
	reduct "github.com/reductstore/reduct-go"
)

func main() {
	// Create a client and use the base URL and API token
	client := reduct.NewClient("http://localhost:8383", reduct.ClientOptions{
		APIToken: "my-token",
	})

	// Remove the bucket with the name "bucket-to-remove"
	bucket, err := client.GetBucket(context.Background(), "bucket-to-remove")
	if err != nil {
		panic(err)
	}

	err = bucket.Remove(context.Background())
	if err != nil {
		panic(err)
	}

	// Check that the bucket no longer exists or is in the process of being removed
	_, err = client.GetBucket(context.Background(), "bucket-to-remove")
	if err == nil {
		panic("Bucket should not exist anymore")
	}

}
